lib/src/main/java/graphql/nadel/dsl/NadelHydrationConditionDefinition.kt

```diff
@@ -0,0 +1,16 @@
+package graphql.nadel.dsl
+
+data class NadelHydrationConditionDefinition(
```

Comment:
Used to be named `WhenConditionDefinition`

Three things

1. Added `Nadel` prefix
3. Removed redundant `When`
4. Mentioned `Hydration`